src/Core__Array.resi

```diff
@@ -29,55 +28,622 @@ let make: (~length: int, 'a) => array<'a>
+
+## Examples
+```rescript
+let someArray = [3, 2, 1]
+someArray->Array.sortInPlace((a, b) => a > b ? 1 : -1)
+
+Console.log(someArray) // [1, 2, 3]
+```
+*/
+@send
+external sortInPlace: (array<'a>, ('a, 'a) => int) => unit = "sort"
 @variadic @send
 external spliceInPlace: (array<'a>, ~start: int, ~remove: int, ~insert: array<'a>) => unit =
   "splice"
-@send external unshift: (array<'a>, 'a) => unit = "unshift"
-@variadic @send external unshiftMany: (array<'a>, array<'a>) => unit = "unshift"
-@send external concat: (array<'a>, array<'a>) => array<'a> = "concat"
-@variadic @send external concatMany: (array<'a>, array<array<'a>>) => array<'a> = "concat"
-@send external flat: array<array<'a>> => array<'a> = "flat"
-@send external includes: (array<'a>, 'a) => bool = "includes"
-@send external indexOf: (array<'a>, 'a) => int = "indexOf"
+
+/**
+`unshift(array, item)` inserts a new item at the start of the array.
+
+Beware this will *mutate* the array.
+
+See [`Array.unshift`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift) on MDN.
+
+## Examples
+```rescript
+let someArray = ["hi", "hello"]
+someArray->Array.unshift("yay")
+
+Console.log(someArray) // ["yay", "hi", "hello"]
+```
+*/
+@send
+external unshift: (array<'a>, 'a) => unit = "unshift"
+
+/**
+`unshiftMany(array, itemsArray)` inserts many new items to the start of the array.
+
+Beware this will *mutate* the array.
+
+See [`Array.push`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift) on MDN.
+
+## Examples
+```rescript
+let someArray = ["hi", "hello"]
+someArray->Array.unshiftMany(["yay", "wehoo"])
+
+Console.log(someArray) // ["yay", "wehoo", "hi", "hello"]
+```
+*/
+@variadic
+@send
+external unshiftMany: (array<'a>, array<'a>) => unit = "unshift"
+
+/**
+`concat(array1, array2)` concatenates two arrays, creating a new array.
+
+See [`Array.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) on MDN.
+
+## Examples
+```rescript
+let array1 = ["hi", "hello"]
+let array2 = ["yay", "wehoo"]
+
+let someArray = array1->Array.concat(array2)
+
+Console.log(someArray) // ["hi", "hello", "yay", "wehoo"]
+```
+*/
+@send
+external concat: (array<'a>, array<'a>) => array<'a> = "concat"
+
+/**
+`concatMany(array1, arrays)` concatenates array1 with several other arrays, creating a new array.
+
+See [`Array.concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) on MDN.
+
+## Examples
+```rescript
+let array1 = ["hi", "hello"]
+let array2 = ["yay"]
+let array3 = ["wehoo"]
+
+let someArray = array1->Array.concatMany([array2, array3])
+
+Console.log(someArray) // ["hi", "hello", "yay", "wehoo"]
+```
+*/
+@variadic
+@send
+external concatMany: (array<'a>, array<array<'a>>) => array<'a> = "concat"
+
+/**
+`flat(arrays)` flattens the provided arrays into a single array.
```

Comment:
A bit circular. What does it mean to flatten?

```suggestion
`flat(arrays)` concatenates an array of arrays into a single array.
```

Reply:
How did this one get through, circular stuff is supposed to be impossible in ReScript.... 😉